components/datatypes/mobile-application.json

```diff
@@ -0,0 +1,133 @@
+          "meta:enum": {
+            "launch": "Launch",
+            "install": "Install",
+            "upgrade": "Upgrade"
+          }
+        },
+        "xdm:isClose": {
+          "title": "Is close event",
+          "type": "boolean",
+          "description": "Indicates whether or not this is an application close event and contains details of the previous session."
+        },
+        "xdm:closeType": {
+          "type": "string",
+          "description": "Specifies the close event type. A value of close means the application signaled that it has closed, either by a user action or by the device. Unknown means the application did not signal that it closed, which could indicate an application crash, a system failure, or an improperly configured application.",
+          "enum": ["close", "unknown"],
+          "meta:enum": {
+            "close": "Close",
+            "unknown": "Unknown"
+          }
+        },
+        "xdm:isDailyEngUser": {
+          "title": "Is daily engaged user event",
+          "type": "boolean",
+          "description": "Indicates whether or not this is the first application launch of the day."
+        },
+        "xdm:isMonthlyEngUser": {
+          "title": "Is monthly engaged user event",
+          "type": "boolean",
+          "description": "Indicates whether or not this is the first application launch of the month."
+        },
+        "xdm:sessionLength": {
+          "title": "Session length",
+          "type": "integer",
+          "description": "Reports the number of seconds that a previous application session lasted based on how long the application was open and in the foreground. Included with close event."
+        },
+        "xdm:ignoredSessionLength": {
+          "title": "Ignored session length",
+          "type": "integer",
+          "description": "If the previous session lasts longer than the maxSessionLength, that session length is ignored and recorded here. Included with close event."
+        },
+        "xdm:daysSinceFirstUse": {
+          "title": "Days since first use",
+          "type": "integer",
+          "description": "Number of days since first launch event."
+        },
+        "xdm:daysSinceLastUse": {
+          "title": "Days since last use",
+          "type": "integer",
+          "description": "Number of days since last launch event."
+        },
+        "xdm:daysSinceLastUpgrade": {
+          "title": "Days since last upgrade",
+          "type": "integer",
+          "description": "Number of days since last upgrade event."
+        },
+        "xdm:launchesSinceLastUpgrade": {
+          "title": "Launches since last upgrade",
+          "type": "integer",
+          "description": "Number of launch events since last upgrade event."
+        },
+        "xdm:installDate": {
+          "title": "Install date",
+          "type": "string",
+          "description": "Date of first launch after installation."
+        },
+        "xdm:launches": {
```

Comment:
Would you consider defining this as "measure" datatype as that is the preferred way of defining events today on the platform. That way, analytics will have a consistent manner to process all events on the platform.